pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/SegmentMetadataImpl.java

```diff
@@ -88,6 +89,11 @@ public SegmentMetadataImpl(File indexDir) throws ConfigurationException, IOExcep
     LOGGER.debug("loaded metadata for {}", indexDir.getName());
   }
 
+  public SegmentMetadataImpl(File indexDir, String pinotHadoopJarVersion) throws ConfigurationException, IOException {
```

Comment:
Modify segment creator metadata builder to add version to metadata. 
this change won't persist and won't be available for reading on servers